src/RAM/Science/components/ClosedLoopControls.tsx

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable prettier/prettier */
+import React, { Component } from 'react';
+import CSS from 'csstype';
+import { rovecomm } from '../../../Core/RoveProtocol/Rovecomm';
+
+const label: CSS.Properties = {
+  marginTop: '-10px',
+  position: 'relative',
+  top: '24px',
+  left: '3px',
+  fontFamily: 'arial',
+  fontSize: '16px',
+  zIndex: 1,
+  color: 'white',
+};
+
+const container: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'column',
+  fontFamily: 'arial',
+  borderTopWidth: '28px',
+  borderColor: '#990000',
+  borderBottomWidth: '2px',
+  borderStyle: 'solid',
+  padding: '5px'
+};
+
+const row: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'row',
+  flexGrow: 1,
+  justifyContent: 'space-around',
+  height: '30px',
+  margin: '10px',
+};
+const column: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'column',
+};
+
+const packetNumList = [0, 1, 2, 3, 4, 5, 6, 7];
+
+interface IProps {
+  style?: CSS.Properties;
+}
+
+interface IState {
+  packetStream: string;
+  currentPacket: number;
+}
+
+class ClosedLoopControls extends Component<IProps, IState> {
+  static defaultProps = {
+    style: {},
+  };
+
+  constructor(props:IProps) {
+    super(props);
+    this.state = {
+      packetStream: 'off',
+      currentPacket: 0,
+    }
+    this.updatePacketStream = this.updatePacketStream.bind(this);
+
+    setInterval(() => this.sendPositionData(), 100);
+  }
+
+  updatePacketStream() {
+    switch(this.state.packetStream) {
+      case 'off': {
+        this.setState({ packetStream: 'on' });
+        break;
+      }
+      default:
+        this.setState({ packetStream: 'off' });
+        break;
+    }
+  }
+
+  sendPositionData() {
+    switch (this.state.packetStream) {
+      case 'on': {
+        rovecomm.sendCommand('GotoPosition', this.state.currentPacket);
+        break;
+      }
+      default: {
+        break;
+      }
+      
+    }
+    
+  }
+
+  render(): JSX.Element {
+    return (
+      <div style={this.props.style}>
+        <div style={label}>Closed Loop</div>
+        <div style={container}>
+          <div style={column}>
+            <button onClick={() => this.updatePacketStream()}>{this.state.packetStream}</button>
```

Comment:
Make `packetString` a boolean and generate the display text based on that. `{this.state.packetStream ? "On" : "Off"}`